{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: Biobase\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "\n",
      "Attaching package: 'BiocGenerics'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:parallel':\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Loading required package: SingleCellExperiment\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: 'S4Vectors'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:base':\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "\n",
      "Attaching package: 'IRanges'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:grDevices':\n",
      "\n",
      "    windows\n",
      "\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: DelayedArray\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: 'matrixStats'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:Biobase':\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "Loading required package: BiocParallel\n",
      "\n",
      "\n",
      "Attaching package: 'DelayedArray'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:matrixStats':\n",
      "\n",
      "    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    aperm, apply, rowsum\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'monocle3'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:Biobase':\n",
      "\n",
      "    exprs, fData, fData<-, pData, pData<-\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(monocle3)\n",
    "library(ggplot2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "cds <- readRDS('output/monocle3-metza-welch.rds')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in library(MatrixMM): there is no package called 'MatrixMM'\n",
     "output_type": "error",
     "traceback": [
      "Error in library(MatrixMM): there is no package called 'MatrixMM'\nTraceback:\n",
      "1. library(MatrixMM)"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "q = plot_cells(cds, color_cells_by = 'dataset',show_trajectory_graph = F,group_label_size = 2, label_cell_groups = F, alpha = 1)\n",
    "\n",
    "ggsave(paste0('plots-submit/welch-dataset.png'), q, dpi=600, width = 6, height = 5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAAAUVBMVEUAAAAAsPYAv30Av8Qz\nMzM5tgBNTU1oaGh8fHyMjIyVkP+ampqjpQCnp6eysrK9vb3Hx8fQ0NDYkADZ2dnh4eHna/Pp\n6enw8PD4dm3/Yrz///8XekGgAAAACXBIWXMAABJ0AAASdAHeZh94AAAgAElEQVR4nO3di3ab\nuBaAYTeZuGnTThu3TT28/4OOb6AtIUDA1g3+b53TSXyJPdP8SyAEPjQAVjvkfgPAFhASoICQ\nAAWEBCggJEABIQEKCAlQEBLS8f7nVdw3A9QqIKRHPnZEjGSAMB3EsSEkYELwpp2zVUdIgBAe\nktlF+ueCkABh3ohkhiVCAoTwWTv7K0ICBEICFLBpByiYFZKYuSMkQJi3smHe84DdWBoEIQEC\nIQEKCAlQQEiAAkICFBASoICQAAWEBCggJEABIQEKCAlQQEiAAkICFBASoICQAAWEBCggJEAB\nIQEKCAlQQEiAAkICFBASoICQAAWEBCggJAx7eXkZvzvVGykfIWHYy3hJlztp6YGQMGKyJEJ6\nICSMGUnl5SHp+ykWIWHMcCovhCQREsZNhkRLV4SEaZ5UXgjJQkiY9qjlXkz75+2fj9tJiZAQ\nQIT0InNixqFFSAggajFDUZcUh5MICUGsceflng4hSYSEIF1KZmdJ3pnnTRWEkBCk2xky41J7\nM5MNDSEhhNioazyTd3RESAhhl2LtKzX9r3eJkDCTmGSQB5cIKenzUD0TUtP+g807QsJs5oDS\n4zv2kwgJ85kDst0NO6+oISTM1c2CW/XsPiVCwjxiybc9KO07JUJCMLE2qNu8M/cQUsrnoWLm\n6KubDSEREmZzT53gnCRCwgziQKwTEiclERKCmQOx3bSddX5f1jeXGSFhrrYicVLF7jsipNX+\n/Mn9DlITS8Dv37KPREjr7SgkfyuEdEVICGavVpU3y629fSIkhHNDEstW937tBkJCuNFrNex7\n446QsJx9CvqOxyNCwhLuyebdMdr9lkRImK9/rQZCSvw8bIEMxr52caY3lB8hYR1xAWNCSvc8\nbE23zmHHFTWEBCW7Ho4aQsIcL/2do3ZA2veGHSFhjt7pfM4uUq73VQBCwgz9WLrL2u06I0LC\nHP5aGI4aQsIczplI8g5CSvs81MyEtPdueggJ8/UGILoiJMznm3LYOUJCKP/O0eBd+0JICNWv\npXfp4v0iJCxnf1LSrhESVmDLrkVIWIGCWoSEBfofRpHpjRSDkLBA/wr6Gd9MEQgJ6xDSDSEh\nhF2K52PGkr6bAhESQnhD6k4wZ0wiJCzwYj4cyVyNK/ebyouQMFu3WySuaZf7PeVGSAjkXBWS\nkCyEhEAv7py35+v9IiSEsi8hJD5GlpAICSHabLpvrIQIqSEkBLA/bKL7motwCYQ0YUcfETvk\nRYTk/ZKUCGkSIVlXyO+PTCwBvyEkTBFjkPuRYoxHLULCFHtjzhqJCKlFSJhiT9M1zlI7Qroh\nJEwSe0i+uzK8o/IQEiYxrTCNkDCFSe4AhIQpRBSAkAAFhIRlGKUshIRlCMlCSIACQgIUEBKg\ngJAABYQEKCAkLMbEnUFIWIyQDELCMlRkISQsQ0gWQgIUEBKWYUSyEBKWeVz+G3eEhGUIyUJI\nWIaKLIQEKCAkQAEhAQoICVBASIACQgIUEBKggJAABYQEKCAkQAEhAQoICVBASIACQgIUEBKg\ngJAABYQEKCAkQAEhAQoICVBASIACQgIUEBKggJAABYQEKCAkQAEhAQoICVBASIACQgIUEBKg\ngJAABSFBHO9/Xsx7HrAbAUHc+zF/hD4P2I/pII4NIQETgjftCAkYtiSkfy4ICRAYkQAFhAQo\nICRAASEBCggJUMDKBkABa+0ABYQEKCAkQAEhAQoICVBASIACQgIUEBKggJAABYQEKCAkQAEh\nAQoICVBASIACQgIUEBKggJAABYQEKCAkQAEhAQoICXV7usj9HhpCQpVu8dwCeiIkYKlHSK3c\nb6chJFTp0U45HRESKtRu2ZXR0A0hoT63hgqqqCEkVKicDTqDkFAdQgIUlJcRIaFChARsFCGh\nHuUNRB1CQj0ICdg2QgIUEBKggJBQsXLmwQkJFSMkQAEhAWt1Z8mWgJBQo6IiuiIk1IfV34AC\nQsK+fP36VfkndpcOUv65axFSZT59Mn+WL1JIBSKkGohubl9+qiWkCCUVipBq0HbTDkc3Od9Q\nsK97KYmQqnDtpquntpB2URIhVeGT0X2d+z2F2UtJhFQFO6SmopBUN+5KnWloCKkOn6T229xv\nKlRgSSEz2oSENawtOhFUHQI37ggJsfnGo2o6agLHpPKOsc5CSDXwTDXUFFLQ0SRCQnS94aiy\nkJr1IZWeGSFVwB2Q6lkiNAMhIS5ngqHGASnI2ClGpWdESBVw9o4ek3jb8zTyyWGEhNXkAFTl\nDpIwtq80tm1HSNDQbtQ9vt5sSAO9lJ8RIdXB3qKrt6NRw5+sTEjQ4SwM2mZJT2M7ScUjpGrI\nibvc70VfiR9nOQchVaPyeYYJT+3st1tTJXURUlU229EtGO9eEiFB33ZDeurYt+Z6P3MRErK6\np/L05IbU3p7vnc1DSMiqu07dU7ttJ25P6OD/jf5xDH3+0tdd+DzA9cjn/v9cY9BASAM3ex64\n9HUXPg9w3Yciz60pERLqZw1EaS9K/PHlcPze3Iu5V3P789/j4fOP69e37/6+HQ5vf2/3/T6+\nen8OISE7Oxox6RC/pr/Haytf3JC+X289/GhDuj3o8+2+18Ob9wcRErJzckkZ0vdLF7/aiExI\nh8PH5eZjOzwdvl8feRuhvg/8IEJCabq5uwTLhj4f/t6/sEM6Ht7eu5svD7p9+eUemN/OQ/rz\nJ/c7wE3XSzcJ3q10iPq63WSCHdL7ZWPu80d7/+FhbO5h3yH9IaRCiJAeX3ZHlqK+7kBITfP7\n8+H4i5DC0FEpnFk7Mxql37T7aGP50d30ufttJyQvQiqFnF54sibuIvt++N78bos5Hn42f1/v\nX/263NxONlwf1Pw8vBLSADoqh1kiJNcMRffRzmx3k97/munvf683H9s58sNvQkLx7DWrj2Ep\nwev+voxAbx/t0HO8xNN+dbx0dNm+uy62+3g7HF4ve0yEhOJZa1brO1t2tyGxWVeYJ3e7rq6U\nCAmlkCcljVx1tUy7DQnleXJSqgkhoRiEBKiotKKGkFAUQgIUVNsRIWW03WtrLUdImI2QfGqs\nqCGknAhpQwgJUEBIgAJCAhQQEqCAkAAFhAQoICRAASEBd+erpU8mJODq3Fr2dEICGtHRwpII\nCbA68pc09QtPSMB0SJMfOEZIgN2Rp6QDIxIwbSokNu2AAIQEKCAkQAEhARqY/gYUEBKgYWpl\nAyEBISZWCBESEGb5QruGkAAVhAQoICRAASEBCggJUEBIgAJCAhQQEqCAkAAFhAQoICRAASEB\nCggJUEBIgAJCAu5OV0ufTEjA1am17OmEBDSio4UlERJgdeQr6XCYuvg3IQFTIR26PwYREmB3\n1C+JkIAQEyHdEBIwgZAABQEhMdkATCEkQMPajggJaKZDmvx9JySgmVrZMP3rTkjA1WhHh8ml\nDYQE3C1faNcQEqCCkAAFhAQoICRAQXgQx6sFzwN2YEZIC58H7AAhAQqCg7A7IiRACg/J7CL9\nc0FIgDBvRGKyAfCaFwQhAV6EBChg0w5QMCskMXNHSIAwb2XDkucBdfh2tfTJrLUDrr61lj2d\nkIBGdLSwJEICrI58JXERfSDEeEhc+xsI8e3baEmEBISYCOmGkIAJhAQomAyJyQZgGiMSoGG6\nI0ICJjH9DWgY30Pq/hhESMDV6IYdKxuAUMsX2jWEBKggJEABIQEKCAlQQEiAAkICFBASoICQ\nAAWEBCggJEABIQEKCAlQQEiAAkIC7l6ulj6ZkICrl9aypxMS0IiOFpZESIDV0UBJnCELTJoM\nadap5r8+Hz7/CHoaIWFLXl4mSjrMGZF+XS/xcHi93UxI2JGpkA6zNu1eD9+b5v14LYmQsCe6\nId3r+biWREjYk4mQDs2CkC4lfSEk7Mp4SAHXh7Tufrtu2l38PrwSEnZlfEC6G/0B8t6P9sE/\nJ68rSUjYFN3jSB9vx/sXv14JCbsyubKBA7JAgKkVQoQEBFm+0K4hJEAFIQEKCAlQQEiAAkIC\nFFhB/H49HN4+5j8P2DsZxO/7Sojfc58H7F5vrd3b4W3u84Dd663+/ns4zn0esHv90ygm16v2\nngfsHiEBCggJUEBIgAI7JGHG84DdIyTg7ulq6ZNZIgRcPbWWPZ2QgEZ0tLAkfxC/vk8dlSUk\nbMnT02hJ864i9PD+djxMLm8gJGzJVEiTP8B9xPvbtb2397nPAyr29DRe0syQ7hUdDn+nX5mQ\nsCETIQX8trvT35exKOiILCFhQ6ZCmj4gZIf05W8TuLRh4jHfvgX8DKAQISNS+GSD4ohESKjJ\n1D7S1YxZO8V9JEpCTaY7mhNS083a/Zp64emQKAn1CJi0y3Ec6dvNxI8AyjF5GGnRtb/Xrmx4\nhHRt6dYTUaF0oxt282bt5picbPgmYur+ICiUa/lCu8YO4qh4GkUXUvPI6Rube9gwGcQX1fOR\nzJAkrHqvQLFkED8On/+dnK7zPM9P1GNt5gEbZH2G7HW67vD2M+AwUsi+lRmEZFDdnfPfK1As\nJ4hf/36+tPT67+Rli0MmKbqNOTukb91kHrAV/SA+frwedM5HGphtYFcJ2+MN4u8XnYuf2FN3\n7Cdhu2KOSE1/xo65O2yTfx/px+SHJAUeyLU6um/hMSZhg3qzdse3d6VZuyt3loHDSdimeMeR\nbqz5hbCQCA0Virey4cE7Io2VQkioUKy1dkY7GokjtAOtiPmI4J8OFCH+lVa/OXzLwNuE5Iw5\nUJEUlyx2Imrajb1v4hHuYxe+LWCx56ulT/Z+GsXrj1nPm9KW0U1/d7X0ByYm9pDHc2vZ0wc+\n1uV1zvMmdSfLunMOni283mgFJPD8vK4kbxDvx8PPJc8b4Oz7dKPRUC6EhNSen0dLWnqq+fvh\ny8QLL9tHEhN4I9PchITUxkNadhWh282qn9gnByP5/4E5cEJCYs/PoyWVEpJz2nkjQ5ITdkza\nIY+QkMalCandXOsm6uyz/MRDxjb5gEgmQypiH+nK2j8SwVh9cdof8pgKafoKkSlm7a6cdh61\nfLMGKns+AkhGdx8p2nGkq3aOof2mLagxm31dYIxKSEx11i7WyoYbO46uGnOg1gpp7k8HVok1\n/T1pVUjODhMTdshu/Hhs98egZCHdmF2j9htrNCIkZDO6QijbRfS95HFZa8ZBzkIsfD/AWssX\n2jXJQ2raLTlrtqHpVoYDdUq9aScGHecbphhQsRwhWZt38p6GCTtUKm1I4mhSN8tg7mF5EKqV\nOKTGXsfQhmQfRgKqkzqkO+swbCOPKAFVSh+SPGZkTqxgtgFVyxeSXAbeEBLqlmvTzhxOsk7t\nIyXUKds+krPGof1m1Y8FcskTUtOY+Tq5v7T2hwKZZAvp4ZtD6ccCaeUOyS1J68cCSWUPqeED\nyLABJYR0xS4Scvt0tfTJpYTUsF4VWX1qLXt6QSEB+Xz6tK4kQgKsjjwlBXyIJSEBUyHdEBIw\n4dOnyZIKuvgJUChCAhRMh6T9qRJrnwcUiJAABYQEaFjbESEBDSFpOJ1Oud8Csls3Z0dIDSHh\nZvxwLCEFOJESmu2s/s5oLCQiQwBC8jP5EBICEJKfDImUMImQ/AgJsxCS3y0eCkIoQhpBSAhF\nSF4khHkIyYuQMA8h+dARZiIk1+mu/Sbvm0EtCMnlC4kZcEwgpB6rpHtLpy4oxih4EZL0OHrU\n6W7r7mdsghchSb6Q3EcQEjwIqedkcSccCGmzvl4tfTIh9TwCeuwbsTG3E19by55OSD1mYuE+\nucAEwx58/bqupPAgjhdLnlcdM/U9trOETfn6dbSkyWvohwdx7P6Y97z62PtIJ6bq9mA8pEP3\nxyBC6hPDULejlPs9IaqvX0dLIqRl7Bk7Otq+PCH9c7GTkLoT/Ghp0xiRVLVLGdodo5OYwiOk\nLZsIicmGebo1Qb0VDmzebRsjUhyEtDfM2qk7mRFJ3kZIm0ZI6uQRWM7v243x47HdH4NY2eCS\ng484liRuwBaNrhBSnGxQel75eq2IM/s8q8GxGcsX2jWE5NMvxTrJj5DQR0h9njHJnndI+3ZQ\nA0J66PVhbrAOzjKBBx9CehgMSS68a1NK//ZQOEJqDYck1w5xehK8CCmMOPGckNBHSIOc63D1\nr4oCdAhpkB1SQ0gYQUgOecUG+9YTIWEQITlESE032e1u2TFzBwchDeiH1IiKuiXiwA0hDRCL\n65qTPSJ1J9FmfosoCCENEEeR7AFJDFFAi5DmsFLK/Wag7M/V0icTUghn2R0bdhv0p7Xs6YQU\nwll2Z0JiBm8r/vxZVxIhjbIakbN25hAtIW3Bnz+jJXGG7EonMxZ1w485HMtx2c0YD0n1mg06\nz6uMFdLjC7NpJ0Kiqar9+TNaEiFpOYlrR8q1DuJ+SqoYIaVx6kKSu0mNs9CBkqpFSLGJy4HL\nI7LtVVBEQQxKFQsIaWK2gZDGmbGoEcOPZ+vuREgVmwjpOmvHiDRXb8q7+6LbRTKX5eqWEBFS\n1Samv68IaSbnzFh798jcIibCxQ1Z3jFWY/o7Lrn1dvJr5L3Z3ihWGj8e2/0xiJBm8Zwpa2/6\noVajG3asbFjLDEciIjnlYA4xEVLlli+0awjJp3f5oKab6jbL605ii8/zPOwMIfXJkUXOcbuX\nEmragUqucEj9ZlEGQvJwc7CnG8xMt5gQH3gm9oKQZnBnGNwVrXS0X4Q0j3sRIXsSnJB2i5Bm\ncE6UFV8S0N4R0gyPiQcnJHsZUd53iFwIaQaxUdebyGvvz/n+kA8hzdLtHNl7RSfZFfaIkKZ1\n5ZjTy8UM+P0RXOlu5whpmhPS/ab+4aN9hHQ+n3O/hSIR0jTrMJFv1nsvEV0Rkh8hzWCNSvsM\niY4GEFIAuT61G4ucCYZ9pHQ+U5IfIQWwFnrby+32FdJ50yH9d7X0yYQUzhwt6pLaV0ebDum/\n1rKnE9I058yj3vog+ZiN20FHC0sipGn2RF23umF/6+w2PCD9999ESfdf+OFTzglpmhlz3NWq\n9tVRNu4e0U5DOpjrn/h/9QlpUm/Mkdt13ffbdx+MthnSf/+Nl3RoCGk1Mwh13/dPON+BPYfU\nEJICZ9G32cbznGu+YY+OCMn7Ewhpmm+W295F2n5I992j80YHJEJKwTfgyDmHDG8ptXaeYR+T\ndoQUhXfLzZrIMzdsVZvPRjNqQqe/CWmFoV0gcTCp2fTQZI9DG22JkOIbDqmRmW06pO1v3U2t\nbCCkcAP/CUdCsi/gsGlnIfd7iWJihRArG4Lc/vP5/xuOzW6LifENO7fTdZsOidXfGkb+A/Yq\nkTecTs3WJxse7Yj/bTWkVQhpijckcePWFzeYIagtipA8dh/S1HDu37ATy4W2v7ChK2fjm3ar\nENKikMS9W5757s12k9EQQpravxzrRB6X1X1bpTg7W3aENICQpidq/JGYdazdRPj2ajLpyOlv\nWurbfUhjE3ajzIFY+8xzvXeWn2imnWggJD9CWupkzdx5btyA7vhRu3NERUMIKYinj34zZjza\nSk7n7vRytuomEFKQ6ZDMTpK4eGTV5P4RG3VTCCmI/1iSXNRgD0db2F2Sy1S7cSn3myoWIS0m\nQjKLGzZ2/rkzUUdIgwhpsVPv+t9NV1P1h5a6ctiiC0NIa8jtOXsta+UhnZ1lQVnfTBUIaR15\nNKm9xfxZLbujrG+lEoS0kn/jrvJ5O/s4LCkFICQd9shU91SDO+vNHEMAQlJx6oXUuPtN9TDH\nYDn5KBghabCm6bo1d7WGZE6FpaNghKTBHDo6+T7NrzbiECwCEdJa3dhzMppaN+3kKRO530td\nCGkts4zBzqjK8UgsC8r9VipDSArsbTq5erWqmNqPbaGjBQhJhwmpl1U1ujOOCGk+QtIidpHa\njb2TGJpqQUbLEJIWa6bB2muqJiT2j5YjJC3iWijORLh9tkWR5KfxEdIShKTDOgrrdnSyiipR\n9/GwDEgLEZIOE0o/HnuLL/c7HcY0wwqEpMO6VKS1p9Sd/mduKBPzdWsQkg55qUhrS866FErp\nIeV+CxUjJCX2tIJ12Ugzz1BuSAxGKxGSEiskGYx9qaFSd5PIaCVCUuMLSeZkLwwvDR2tQ0hK\n5Py33VT7D3NneQeUzMctYxlCUiJbkSf1OSdUdDPhZWGp6lqEpMU+aiQm7cwnwDRFTjeI66nm\nfisVIyQtzuFX+8qrzsxd3nfqMIu+c7+TmhGSovYEJGuFg3WBofaAbd732UdGaxGSHnE2n9y6\nu9/VjVbtA4sgTkHK/VYqR0h63KV2g5dDKWbjTn7gRO73UjlCUnbqk9dwkIdn83sEREbrEZIa\nk4sZmqxkSsmnxcceKSIkNc50nLM514hNv0KS4uJ1ighJjZOImW6wJhjcaYhsiEgVISmSY5B7\nVLY/LZ4X+0W6CEmRdcRIlCSGpkIyIiRthKTIWsvQFXS/RzQmph2yFUVHyghJm7V21TqPwnPt\n1TwhcdxIHyHpModg5fSCuVc+LOTnRfmFpyN9hKSrF5K7SyRuDfl5Zm5N8XefkPQRkj5raaoM\nqbdlF8Z8/uT633/rJ5CTHkKKyD0eu3iPqB1BFv3mu+1Yn7S88A3BRUgRyRNmexN2c6z6hZdP\nlh+tzNogTYQUmXuRhhUhaWzYneV4tPbHwSCkSMS6IHv9d2PfFKi9DGrYL794mDMesTUXCSHF\ncRIhNfZR2u77OT/vMd+wKKTuOzKKh5Di6Geybtn3uStpKgUxu+cOYwxI8RBSVIunF1xtENMt\niHbczUGmF+IhpHhOK+bpLGJkCdjAsx7s3LP2nWAIIWlzzkjy3D5bfxMt/MED8w5QRkjarHas\nxXULf+DZrGxob/A8Rj7crIUwG3nyQRSlj5C0+Remrh6PHl/d/+F5jPOEx43ymWaYIiR9hBRN\ndz7Fuh9z7oX0uNn30O4Z/Z9htvnIKAJCimxowiG4L+d338zHeR9qPcp60pmQYiKkyNqzKtwz\nKeYMVP6QfBt4I3tBBBQVIUU2FFKY4Z0ae/7hcQupZENIkVkn9s3eX/KmMTAoTWdEavEQUmxy\nYdDckLpZa2tkuk8cyNMh7Cf4fsjjK0KKhZBiWzNr5w+peezw9Lf7vPPjxJMCISWkfM0gzzDl\nP9DkWy4EXYSUxuyZulHu7Fy7jMHZeJMbf0ORQQchpaF3ddV2eJGrfkRF9vlH8k+ZE7QRUgy+\naOzblmUlI+rNffdDch9IRvEQUgzTmYSH1J9L8D9Krlow23ScaJ4IIZWut3x7IAYTklksLoJi\nfVBchBTL+sWqA9+JGryLvrsJPF9GhBQJIcWyPKR2WOm+du/u9oOckKyvu26YZUiBkMrT/fY3\nvYluMV3XK0M+wyz3NgdvKSkmQiqQ77feCclTUj+k9na26uIjpGjWbNt59oOs2YPRKbhuODIP\ntEYnREBI0aw9u7z72pwzYTb4vBmd5bDVbdCZo0sMS/EQUnl8W23++bj+05whyx2P4r7xPSOk\n4phJOc9dI6OKaaXrrZERkVFMhFQKa26g8Q8fZ7Ms1RTT+wmNNQzZO1mIhJBKIX7fxW7N2MPE\nqu/eLWY4SvLmQUhFGl0M5H1w105j/YNdo0QIqRDmt33w935olUN71Kib2mva8ajhSGwq4UEc\nrxY8D0HkDs5AVp6dnW5yuzFRiRuYYkhmRkgLn4d57JD8h4vk0dneFEP7tXUnIiOkkvSn63zz\ncnZI91vsiTp3JKKn6IKDsDsiJF1ixYJzk3dW2932E/f0mmG6IYnwkMwu0j8XhKSqV417MMnJ\npbGHHjPd7f3RdBTfvBGJyYbIxOIEexNNTG6LW87yu+GBh5DimxcEIUUmQ5JBedf6BIeE+AKC\nENPehJSM2OUxg5PchLPnFpjozoxNu/J0e0f+kDwfGWZ/hwxmhSRm7ggpArForttU89RzdkIS\n0wnElMu8lQ1Lnodpckenm1c4y6L6CVkhmcmJbP8OO8dauzgCf6Pt+QMTkj23MMCeBCehrAgp\njsBfbCuWx/R2Y7XidNXfVZrxaoiHkOKY9avt2wOSVbU/UdTUWCEhO0IqhW/bTYR0Ppt57/v3\njbOkjqZyIqQ4FvxWmymGx3hkrZNrtwGdM4/ESxFSToQUx4KtLmuXxyRl7m7/7O0gqb1rLEZI\ncYSGZJXSjkTDIfWmJBrxT/cHIiFCiiRwQJKPsg4b+R7UleOZmyCkvAgpkrNYbzD4EP/zzE6Q\n80AxwWA1NPzTkAohxXFeGJLYqOtvHbohTf40JENIcZxDSvI+zzx9bDfLHpHEph3yIKRIFoZk\nPX38AXKegYhyI6RI5s/aBd3uecD8iXboI6RY5s/aObdbd43sEHVzdvSUESHFsm6gCAnJO3WO\nPAgpltVbXONP71YJOTUhD0KKpTdoLPsJE3ee178MNBBSLEs3tsxGXf/Jnj0hQioDIcW05Bdc\n7B31ns6UQrEIKSa1vSTqKR0hxaT2+09IpSOkmJiQ3g1Ciij+mgNCLQUhRTQYUsjv//RjQk8e\nRAKEFNVgSNMBeKfsrK/pqCCEFNPIkOTeETIA2V9TUUkIKSZPSEPHiMZGF/9dhFQSQoqqfy74\n4K8/IVWNkKJadVGFibV2hFQSQooqzqJVzpkoDyFFNrzMZ8156GzXlYaQIjObYHOmF9xH9r4n\npMIQUmzDvSwNia26AhFSdIuvbzL0LEIqECFFNxzSkh7GL3eHXAgpusHf+4U9EFKJCCk66xr3\n7j3dV01gWCwOKhMhJaAUEoePCkZICfQu4WhdAmjGj2GzrliElIL7GSzmMkEz5w4IqVSElIZ/\n687kFXiGEiGVipCSOFtjkHPzxCe4iEcTUrEIKQmxTWeH1JxDMhKTEnRUJkJKQikkBqRiEVIS\n4qP1eiEFHhZ67EkRUpkIKQ0rpC6JOWGcWWVXMkJKQ059y4RmnBHBIruSEVIq/g7mnFpERgUj\npGTEmjsxOk08xXp6vPcmPT+neZ1tIaS05ApWM0ZNrw9PsF13CejW0PNV3JfaIEJKyZ7p7o9Q\n48+M+tZEQJS0ACGl5Fyda/yUCDe6FCF1XxDSTISUki+F4noAAAW/SURBVBNS445Rgw+OvYd0\n26wTGRHSXISUjFjG4OwnjV5j9ZziAJJo57GfFPXlNoiQknFCEjc31haefY5F8JLW+Z7dfKz7\nIrzglhFSMmLxtu+ECvfLRz8RT4vtQmrn625/3v/Pxt1MhJSKGWEGZhja0UesJzqHn2OxQNeK\n9UUXEiXNQUipTK70lrtP1pq8JAdixZxd+32Kl90MQkpoMCTPaX9nIdb7MdtzIqT2LkKahZCS\nGuiiv4fULvaOOh4932e9n53bnhtCmo2Q0poOY2rZkKLnxyyDbEYub4j+BjaEkNIyU3Hu7eIR\n6d7OPRd3q669K937qB8hJXWeDimp57YkcUt7R5Y3VC1CSqnd5ynmzCJC0kJIKcWePZivt7KO\ngJYhpKRmhJTu8JFY4EBISxFSUjPqSHgctr+Bh7kIKbnAyzQkHpHaY7NmlRBlzUFIyYWFlGpf\nyspIDE39aXGMIaQMQiJJF9Jjy84amhqGpJkIKYOyJu66lkQ5niUPGEVIZUrdmthXEgURUjBC\nKobVTuLjTc9OSPfbEr6B6hFSMTKG1K64k0u/ObdvFkIq0JzrGKuQsw2NOyeOEIRUoPTriMSm\n3bM1+4BAhFSk1CW5IT2zhzQTIZUp/Qx5t5/EJSKXICTcmctwEdIChATDmgSnpDkIqUR5lj7I\nHSV2kmYipALlOfnPXnFHRvMQUmmynURrVjQQ0nyEVJrsJ6PT0RKEVJzca8NZ9r0EIRWk+8ik\n3G8EsxFSQQipXoRUHEKqESGVh5AqREiFKe4SkghCSIUhpDoRUmHoqE6EVBhCqhMhFcR8dCxq\nQ0gFif2ZsYiHkMpCR5UipLIQUqUIqSh0VCtCKgoh1YqQSsJcQ7UIqST+TzxHBQipJAxH1SKk\nkhBStQgJUEBIgAJCAhQQEqCAkAAFhAQoICRAASEBCggJUEBIgAJCAhQQEqCAkAAFhAQoICRA\nASEBCggJUEBIgAJCAhQQEqCAkAAFhAQoICRAASEBCggJUEBIgAJCAhQQEqCAkAAFhAQoICRA\nASEBCggJUEBIgAJCAhQsDimxf1K/4N5edcv/qqrFDAWR4kUU/MOrbvBFM71qDITEq+Z7UUJK\nbk9/z4RUoVpCAopGSIACQgIUEBKggJAABYQEKKgjpOP9z4vEr5v+JTP8WzaZ/kW7V078wlFU\nEdL9P7X5I90LJ30185LJXzfHv2iuv9Q4agjp2BBSildN/IqElF6m/+Y5/oKz/GZl+U0mpPRy\nhZRrzyHHflmeUZCQkso5IqWe38jwmrlflJBSyfnffA8hmVdO/XqElEK3wZH8v7nY1CGkmK9H\nSEmxabfpFyWkVDKGtJvfaSYbVqkopEwrG9K+YsaVDclfM9srx1BHSEDhCAlQQEiAAkICFBAS\noICQAAWEBCggJEABIQEKCCml7oMRrl8cv/z4uH3z8ePL8XHrof3n1dtv+dzv/FWVjL+dlKyQ\nrqncvnl7BPR++ef7/d4HUdL3NJ9OgoX420nJCenzY7nZ5/vtb4cv97QeD/t+eO2e+ZboY36w\nEH87KTkh/Xv4dfn61+Wfh/uNf9svnIc3x+MvQioafzspOSFdErp8fcnpdvv7ZTh6u23b9UP6\nLr9BgfjbSckJ6bJRd/n68+F++zWi99u23f1hf98e+1DOc1Ei/nZSckN6O3w0H5dcbrfLPx4+\nfM9FifjbSckN6f3wo/lx+Hn79v0Rz3sb0tGe/iakovG3k5Ib0t/Da/N6+Hv79q09ejTQDCEV\njb+dlF7vx4kuo8/rvYxLRe2Xx0tQlx2jw5GQasTfTko/DsdrSe/HyybdrYwfhy/XmbvLl7+6\no7O/CKlC/O0k9frYfrseab2W8XFfvnD58vvtmNJ1sPpOSBXibyetn18u6Xz5ef3yVsbxuil3\nmwnvLkh5JKQK8bcDKCAkQAEhAQoICVBASIACQgIUEBKggJAABYQEKCAkQAEhAQoICVDwP53R\nBjXPtmcpAAAAAElFTkSuQmCC",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      },
      "text/plain": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "q = plot_cells(cds ,show_trajectory_graph = F,group_label_size = 2, label_cell_groups = F, alpha = 1)\n",
    "\n",
    "ggsave(paste0('plots-submit/welch-clusters.png'), q, dpi=600, width = 6, height = 5)\n",
    "\n",
    "q"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "NULL"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#plot_cells(cds, show_trajectory_graph = F)\n",
    "\n",
    "\n",
    "\n",
    "#write.csv(as.data.frame(colData(cds)),'welch-overlap-metadata.csv')\n",
    "\n",
    "colData(cds)$monocle3_clusters = clusters(cds)\n",
    "colData(cds)$UMAP1 = reducedDims(cds)$UMAP[,1]\n",
    "colData(cds)$UMAP2 = reducedDims(cds)$UMAP[,2]\n",
    "\n",
    "\n",
    "library(Matrix)\n",
    "\n",
    "writeMM(counts(cds), 'welch-merged.mtx')\n",
    "write.csv(colData(cds), 'welch-merged_cell-metadata.csv')\n",
    "write.csv(rowData(cds), 'welch-merged_gene-metadata.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "write.csv(as.data.frame(colData(cds)),'welch-overlap-metadata.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "q = plot_cells(cds, color_cells_by = 'partition', show_trajectory_graph = F, group_label_size = 1, label_cell_groups = T, alpha = 1)\n",
    "\n",
    "ggsave(paste0('plots-submit/saunders-partition.png'), q, dpi=600, width = 5, height = 5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Aligning cells from different batches using Batchelor. \n",
      "Please remember to cite:\n",
      "\t Haghverdi L, Lun ATL, Morgan MD, Marioni JC (2018). 'Batch effects in single-cell RNA-sequencing data are corrected by matching mutual nearest neighbors.' Nat. Biotechnol., 36(5), 421-427. doi: 10.1038/nbt.4091\n",
      "\n",
      "Warning message in (function (jobs, data, centers, info, distance, k, query, get.index, :\n",
      "\"tied distances detected in nearest-neighbor calculation\"\n",
      "No preprocess_method specified, and aligned coordinates have been computed previously. Using preprocess_method = 'Aligned'\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n",
      "  |======================================================================| 100%\n"
     ]
    }
   ],
   "source": [
    "focused.cds = cds[,(partitions(cds) == '1') | (colData(cds)$region == 'SN')]\n",
    "focused.cds = preprocess_cds(focused.cds, num_dim = 25)\n",
    "focused.cds = align_cds(focused.cds,alignment_group = 'dataset')\n",
    "focused.cds = reduce_dimension(focused.cds)\n",
    "focused.cds = cluster_cells(focused.cds, reduction_method = 'UMAP',k = 20)\n",
    "focused.cds = learn_graph(focused.cds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "q = plot_cells(focused.cds , color_cells_by = 'region',show_trajectory_graph = F,group_label_size = 2, label_cell_groups = F, alpha = 1)\n",
    "\n",
    "ggsave(paste0('plots-submit/focused-saunders-region.png'), q, dpi=600, width = 6, height = 5)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "q = plot_cells(focused.cds , color_cells_by = 'dataset',show_trajectory_graph = F,group_label_size = 2, label_cell_groups = F, alpha = 1)\n",
    "\n",
    "ggsave(paste0('plots-submit/focused-saunders-dataset.png'), q, dpi=600, width = 6, height = 5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: cowplot\n",
      "\n",
      "\n",
      "********************************************************\n",
      "\n",
      "Note: As of version 1.0.0, cowplot does not change the\n",
      "\n",
      "  default ggplot2 theme anymore. To recover the previous\n",
      "\n",
      "  behavior, execute:\n",
      "  theme_set(theme_cowplot())\n",
      "\n",
      "********************************************************\n",
      "\n",
      "\n",
      "Loading required package: Matrix\n",
      "\n",
      "Warning message:\n",
      "\"package 'Matrix' was built under R version 3.6.2\"\n",
      "\n",
      "Attaching package: 'Matrix'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:S4Vectors':\n",
      "\n",
      "    expand\n",
      "\n",
      "\n",
      "Loading required package: patchwork\n",
      "\n",
      "\n",
      "Attaching package: 'patchwork'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:cowplot':\n",
      "\n",
      "    align_plots\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'liger'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:BiocGenerics':\n",
      "\n",
      "    normalize\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(liger)\n",
    "analyze.liger <- function(cds, object.name, correct, fields)\n",
    "{\n",
    "    objects = list()\n",
    "    fields.values = unique(colData(cds)[[correct]])\n",
    "    \n",
    "    for( f in fields.values){\n",
    "        print(f)\n",
    "        objects[[f]] = as.matrix(counts(cds[,(colData(cds)[[correct]] == f)]))\n",
    "    }\n",
    "    \n",
    "    ligerex = createLiger(objects) #Can also pass in more than 2 datasets\n",
    "    ligerex = normalize(ligerex)\n",
    "    ligerex = selectGenes(ligerex, var.thresh = 0.1)\n",
    "    ligerex = scaleNotCenter(ligerex)\n",
    "    ligerex = optimizeALS(ligerex, k = 20) \n",
    "    ligerex = quantileAlignSNF(ligerex)\n",
    "    ligerex = runUMAP(ligerex)\n",
    "    saveRDS(ligerex, paste0('./output/ligerex-',object.name,'.rds'))\n",
    "     for(f in fields){\n",
    "        ligerex@cell.data[[f]] = as.factor(colData(cds)[f][rownames(ligerex@cell.data),])\n",
    "        q = plotFeature(ligerex, f, by.dataset = F, discrete = T,return.plots = T)\n",
    "        ggsave(plot = q, width = 4, height = 4, dpi = 300, filename = paste0('./plots/liger-', object.name, '-',f,'.png'))\n",
    "    }   \n",
    "    ligerex\n",
    "}\n",
    "\n",
    "\n",
    "#liger.obj = readRDS(paste0('./output/ligerex-','focused-saunders-metzakopian','.rds'))\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "q = plot_cells(focused.cds , show_trajectory_graph = F,group_label_size = 2, label_cell_groups = T, alpha = 1)\n",
    "\n",
    "ggsave(paste0('plots-submit/focused-saunders-clusters.png'), q, dpi=600, width = 6, height = 5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "  CB_3   CB_4   DAn1   DAn3   DAn4  ENT_4   FC_1  FC_11   FC_2   GP_2   GP_3 \n",
       "     7     53    212      2      9     44      1      2      5    634      1 \n",
       "  PC_4   SN_3   SN_4 STR_13 STR_14   TH_2   TH_3 \n",
       "     1   1545      2     13     98      8    871 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "selection = c('1')\n",
    "meta.cells = colData(focused.cds)[clusters(focused.cds) %in% selection,]\n",
    "\n",
    "table(meta.cells$Cell.Type)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "cds.orig = cds\n",
    "cds  = focused.cds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "dan1.clusters = c('1')\n",
    "#dan4.clusters = c('60','25')\n",
    "colData(cds)$clusters= clusters(cds)\n",
    "\n",
    "colData(cds)$overlap = 'none'\n",
    "#colData(cds)$overlap[colData(cds)$clusters %in% dan4.clusters] = 'DAn4_overlap'\n",
    "colData(cds)$overlap[colData(cds)$clusters %in% dan1.clusters] = 'DAn1_overlap'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "q = plot_cells(cds , color_cells_by = 'overlap', show_trajectory_graph = F,group_label_size = 2, label_cell_groups = T, alpha = 1)\n",
    "ggsave(paste0('plots-submit/focused-saunders-overlap.png'), q, dpi=600, width = 6, height = 5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "q = plot_cells(cds , color_cells_by = 'Cell.Type', show_trajectory_graph = F,group_label_size = 2, label_cell_groups = F, alpha = 1)\n",
    "ggsave(paste0('plots-submit/focused-saunders-celltype.png'), q,  dpi=600, width = 6, height = 5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'CB_3'</li>\n",
       "\t<li>'CB_4'</li>\n",
       "\t<li>'DAn1'</li>\n",
       "\t<li>'DAn3'</li>\n",
       "\t<li>'DAn4'</li>\n",
       "\t<li>'ENT_4'</li>\n",
       "\t<li>'FC_1'</li>\n",
       "\t<li>'FC_11'</li>\n",
       "\t<li>'FC_2'</li>\n",
       "\t<li>'GP_2'</li>\n",
       "\t<li>'GP_3'</li>\n",
       "\t<li>'PC_4'</li>\n",
       "\t<li>'SN_3'</li>\n",
       "\t<li>'SN_4'</li>\n",
       "\t<li>'STR_13'</li>\n",
       "\t<li>'STR_14'</li>\n",
       "\t<li>'TH_2'</li>\n",
       "\t<li>'TH_3'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'CB\\_3'\n",
       "\\item 'CB\\_4'\n",
       "\\item 'DAn1'\n",
       "\\item 'DAn3'\n",
       "\\item 'DAn4'\n",
       "\\item 'ENT\\_4'\n",
       "\\item 'FC\\_1'\n",
       "\\item 'FC\\_11'\n",
       "\\item 'FC\\_2'\n",
       "\\item 'GP\\_2'\n",
       "\\item 'GP\\_3'\n",
       "\\item 'PC\\_4'\n",
       "\\item 'SN\\_3'\n",
       "\\item 'SN\\_4'\n",
       "\\item 'STR\\_13'\n",
       "\\item 'STR\\_14'\n",
       "\\item 'TH\\_2'\n",
       "\\item 'TH\\_3'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'CB_3'\n",
       "2. 'CB_4'\n",
       "3. 'DAn1'\n",
       "4. 'DAn3'\n",
       "5. 'DAn4'\n",
       "6. 'ENT_4'\n",
       "7. 'FC_1'\n",
       "8. 'FC_11'\n",
       "9. 'FC_2'\n",
       "10. 'GP_2'\n",
       "11. 'GP_3'\n",
       "12. 'PC_4'\n",
       "13. 'SN_3'\n",
       "14. 'SN_4'\n",
       "15. 'STR_13'\n",
       "16. 'STR_14'\n",
       "17. 'TH_2'\n",
       "18. 'TH_3'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"CB_3\"   \"CB_4\"   \"DAn1\"   \"DAn3\"   \"DAn4\"   \"ENT_4\"  \"FC_1\"   \"FC_11\" \n",
       " [9] \"FC_2\"   \"GP_2\"   \"GP_3\"   \"PC_4\"   \"SN_3\"   \"SN_4\"   \"STR_13\" \"STR_14\"\n",
       "[17] \"TH_2\"   \"TH_3\"  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "cluster.selection = cds[,clusters(cds) %in% dan1.clusters]\n",
    "colData(cluster.selection)$type = 'in-vitro'\n",
    "colData(cluster.selection)$type = with(colData(cluster.selection), paste0(region,'_',subcluster))\n",
    "\n",
    "histo.table = table(colData(cluster.selection)$Cell.Type) \n",
    "\n",
    "ct.to.consider.dan1 = names(histo.table)\n",
    "ct.to.consider.dan1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "write.csv(as.data.frame(colData(cds)[(colData(cds)$overlap == 'DAn1_overlap') & (colData(cds)$dataset == 'metzakopian'),]), 'saunders-overlap.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "overlap.cds = cds[,colData(cds)$Cell.Type %in% c(ct.to.consider.dan1)]\n",
    "ct.view = colData(overlap.cds)\n",
    "\n",
    "\n",
    "#all.in.clusters= ct.view[ct.view,]\n",
    "#as.data.frame(all)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 14 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>region</th><th scope=col>total.cells</th><th scope=col>cells.in.overlap</th><th scope=col>percent</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>CB_3  </td><td>  11</td><td>   7</td><td>0.6363636</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>CB_4  </td><td> 112</td><td>  53</td><td>0.4732143</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>DAn1  </td><td>1044</td><td> 212</td><td>0.2030651</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>ENT_4 </td><td>  61</td><td>  44</td><td>0.7213115</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>FC_1  </td><td>   1</td><td>   1</td><td>1.0000000</td></tr>\n",
       "\t<tr><th scope=row>9</th><td>FC_2  </td><td>   6</td><td>   5</td><td>0.8333333</td></tr>\n",
       "\t<tr><th scope=row>10</th><td>GP_2  </td><td>2632</td><td> 634</td><td>0.2408815</td></tr>\n",
       "\t<tr><th scope=row>11</th><td>GP_3  </td><td>   7</td><td>   1</td><td>0.1428571</td></tr>\n",
       "\t<tr><th scope=row>12</th><td>PC_4  </td><td>   1</td><td>   1</td><td>1.0000000</td></tr>\n",
       "\t<tr><th scope=row>13</th><td>SN_3  </td><td>3421</td><td>1545</td><td>0.4516223</td></tr>\n",
       "\t<tr><th scope=row>15</th><td>STR_13</td><td> 102</td><td>  13</td><td>0.1274510</td></tr>\n",
       "\t<tr><th scope=row>16</th><td>STR_14</td><td>  98</td><td>  98</td><td>1.0000000</td></tr>\n",
       "\t<tr><th scope=row>17</th><td>TH_2  </td><td>  14</td><td>   8</td><td>0.5714286</td></tr>\n",
       "\t<tr><th scope=row>18</th><td>TH_3  </td><td>2688</td><td> 871</td><td>0.3240327</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 14 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & region & total.cells & cells.in.overlap & percent\\\\\n",
       "  & <fct> & <int> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & CB\\_3   &   11 &    7 & 0.6363636\\\\\n",
       "\t2 & CB\\_4   &  112 &   53 & 0.4732143\\\\\n",
       "\t3 & DAn1   & 1044 &  212 & 0.2030651\\\\\n",
       "\t6 & ENT\\_4  &   61 &   44 & 0.7213115\\\\\n",
       "\t7 & FC\\_1   &    1 &    1 & 1.0000000\\\\\n",
       "\t9 & FC\\_2   &    6 &    5 & 0.8333333\\\\\n",
       "\t10 & GP\\_2   & 2632 &  634 & 0.2408815\\\\\n",
       "\t11 & GP\\_3   &    7 &    1 & 0.1428571\\\\\n",
       "\t12 & PC\\_4   &    1 &    1 & 1.0000000\\\\\n",
       "\t13 & SN\\_3   & 3421 & 1545 & 0.4516223\\\\\n",
       "\t15 & STR\\_13 &  102 &   13 & 0.1274510\\\\\n",
       "\t16 & STR\\_14 &   98 &   98 & 1.0000000\\\\\n",
       "\t17 & TH\\_2   &   14 &    8 & 0.5714286\\\\\n",
       "\t18 & TH\\_3   & 2688 &  871 & 0.3240327\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 14 × 4\n",
       "\n",
       "| <!--/--> | region &lt;fct&gt; | total.cells &lt;int&gt; | cells.in.overlap &lt;int&gt; | percent &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | CB_3   |   11 |    7 | 0.6363636 |\n",
       "| 2 | CB_4   |  112 |   53 | 0.4732143 |\n",
       "| 3 | DAn1   | 1044 |  212 | 0.2030651 |\n",
       "| 6 | ENT_4  |   61 |   44 | 0.7213115 |\n",
       "| 7 | FC_1   |    1 |    1 | 1.0000000 |\n",
       "| 9 | FC_2   |    6 |    5 | 0.8333333 |\n",
       "| 10 | GP_2   | 2632 |  634 | 0.2408815 |\n",
       "| 11 | GP_3   |    7 |    1 | 0.1428571 |\n",
       "| 12 | PC_4   |    1 |    1 | 1.0000000 |\n",
       "| 13 | SN_3   | 3421 | 1545 | 0.4516223 |\n",
       "| 15 | STR_13 |  102 |   13 | 0.1274510 |\n",
       "| 16 | STR_14 |   98 |   98 | 1.0000000 |\n",
       "| 17 | TH_2   |   14 |    8 | 0.5714286 |\n",
       "| 18 | TH_3   | 2688 |  871 | 0.3240327 |\n",
       "\n"
      ],
      "text/plain": [
       "   region total.cells cells.in.overlap percent  \n",
       "1  CB_3     11           7             0.6363636\n",
       "2  CB_4    112          53             0.4732143\n",
       "3  DAn1   1044         212             0.2030651\n",
       "6  ENT_4    61          44             0.7213115\n",
       "7  FC_1      1           1             1.0000000\n",
       "9  FC_2      6           5             0.8333333\n",
       "10 GP_2   2632         634             0.2408815\n",
       "11 GP_3      7           1             0.1428571\n",
       "12 PC_4      1           1             1.0000000\n",
       "13 SN_3   3421        1545             0.4516223\n",
       "15 STR_13  102          13             0.1274510\n",
       "16 STR_14   98          98             1.0000000\n",
       "17 TH_2     14           8             0.5714286\n",
       "18 TH_3   2688         871             0.3240327"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ct.all  = ct.view[ct.view$Cell.Type %in% ct.to.consider.dan1,]\n",
    "ct.select  = ct.all[ct.all$overlap =='DAn1_overlap',]\n",
    "histo = as.data.frame(table(ct.all$Cell.Type ))\n",
    "histo.select = as.data.frame(table(ct.select$Cell.Type ))\n",
    "#histo = histo[histo$Freq > 0,] \n",
    "\n",
    "#colnames(histo) = c('overlap', 'celltype', 'cells')\n",
    "#histo\n",
    "\n",
    "histo.summary = merge(histo,histo.select, by ='Var1')\n",
    "colnames(histo.summary) = c('region','total.cells', 'cells.in.overlap')\n",
    "\n",
    "histo.summary$percent = as.numeric(with(histo.summary, cells.in.overlap / total.cells))\n",
    "\n",
    "histo.summary[histo.summary$percent > 0.1,]\n",
    "\n",
    "\n",
    "write.csv(histo.summary, './output/saunders-dan1-overlap.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
